### PR TITLE
Html parser

### PR DESCRIPTION
**Pull request by phorward**
Merge the HTML parsing features into develop.

### CODE REVIEW

File: parse.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+	Generates a dictionary of all to the html5-library
+	known tags and their associated objects and attributes.
+	"""
+	tags = {}
+
+	for cname in dir(html5):
+		cl = getattr(html5, cname)
+		try:
+
+			inst = cl()
```
Comment (tsteinruecken) [inline parse.py]: For security, don't instanciate each class optimistically and then use isinstance, check if cl issubclass of widget. (Prevents you from accidental calling each function that might exist in that namespace

File: parse.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+				for fname in dir(inst):
+					if fname.startswith("_set"):
+						attr.append(fname[4:].lower())
+
+				tags[cname.lower()] = (cl, attr)
+
+		except:
+			pass
+
+	for tag in sorted(tags.keys()):
```
Comment (tsteinruecken) [inline parse.py]: Debug? Double-Sort call

File: parse.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+	# Handle defaults
+	if appendTo is None:
+		appendTo = html5.Body()
+
+	if bindTo is None:
+		bindTo = appendTo
+
+	# Prepare stack and input
+	stack.append((appendTo, None))
+	html = [ch for ch in html]
```
Comment (tsteinruecken) [inline parse.py:130]: General question: what is faster in python? [y for y in x] or x[:] or list(x) to obtain a shallow copy?

File: parse.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+		# Create tag
+		if tag:
+			wdg = _tags[tag][0]()
+
+			parent.appendChild(wdg)
+			stack.append((wdg, tag))
+
+			#print("tag", tag)
+
+			while html:
```
Comment (tsteinruecken) [inline parse.py:208]: Logic-Flow: That's the 3rd nested "while html"

File: parse.py [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+				scanWhite(html)
+				if not html:
+					break
+
+				# End of tag >
+				if html[0] == ">":
+					html.pop(0)
+					break
+
+				# Closing tag at end />
```
Comment (tsteinruecken) [inline parse.py:218]: It's hard to follow the flow here, why are there two "closing tag at end />" sections?

File: parse.py [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+
+		except:
+			pass
+
+	for tag in sorted(tags.keys()):
+		print("%s: %s" % (tag, ", ".join(sorted(tags[tag][1]))))
+
+	return tags
+
+def fromHTML(html, appendTo = None, bindTo = None):
```
Comment (tsteinruecken) [inline parse.py:62]: General Question: Why parsing yourself? Browsers can do: https://developer.mozilla.org/en-US/docs/Web/API/DOMParser

File: parse.py (hunk 1 above)
Comment (phorward) [reply parse.py]: OK I've done a fix for this which is committed soon.

File: parse.py (hunk 2 above)
Comment (phorward) [reply parse.py]: The double sort call should remain, because it first sorts by the tag names, then each tag's attributes. The debug should remain in first place.

File: parse.py (hunk 6 above)
Comment (phorward) [reply parse.py:62]: Maybe this can be replaced later. Currently, our HTML parser should only understand tags and attributes supported by our html5 library, to make app development easier, faster and more reliable.

File: parse.py (hunk 3 above)
Comment (phorward) [reply parse.py:130]: The way it works in Python  may differ here because we compile into JavaScript. In my opinion no requirement to be changed now.

File: parse.py (hunk 4 above)
Comment (phorward) [reply parse.py:208]: Correct. The first loop parses text in front of tags, the second parses the tags and the third parses the attributes in tags. It runs very well even with broken or incomplete HTML, which is interpreted as text then.

File: parse.py (hunk 5 above)
Comment (phorward) [reply parse.py:218]: As explained above.
